zip-0315.rst

```diff
@@ -1,9 +1,629 @@
+  information about whether or not the shielded notes we spent were
+  trusted.
+
+
+Note selection
+''''''''''''''
+
+TODO: consider what we should do when nullifiers are revealed in a transaction
+that is then invalidated. Should those notes be prioritized to be spent soon,
+or should they be used in a note management tx?
+
+
+Expiration height
+'''''''''''''''''
+
+A wallet SHOULD create transactions using the default expiration height of
+40 blocks from the current height, as specified in [#zip-0203]_.
+
+
+Open question
+'''''''''''''
+
+How should wallet developers time transactions to avoid linkability?
+
+* when we roll addresses with transparent components, we have to consider
+  how that could allow linking of shielded components
+
+
+TODO: dusting attack mitigation
+
+
+Network-layer privacy
+---------------------
+
+
+Viewing keys
+------------
+
+What they are supposed to reveal; see ZIP 310 for Sapling (needs updating for
+Orchard). https://github.com/zcash/zips/issues/606
+
+
+
+Allowed transfers
+-----------------
+
+* Sprout -> transparent or Sapling
+* Sapling -> transparent or Sapling or Orchard
+* Orchard -> transparent or Sapling or Orchard
+* if auto-shielding is off:
+  *  transparent -> transparent or Sapling or Orchard
+* if auto-shielding is on:
+  *  transparent -> internal Orchard or Sapling
+
+Note: wallets MAY further restrict the set of transfers they perform.
+
+
+Auto-shielding
+--------------
+
+Wallets SHOULD NOT spend funds from a transparent address to an external address,
+unless the user gives explicit consent for this on a per-transaction basis.
+
+In order to support this policy, wallets SHOULD implement a system of auto-shielding
+with the following characteristics.
+
+
+If auto-shielding functionality is available in a wallet, then users MUST be able
+to explicitly consent to one of the following possibilities:
+
+* auto-shielding is always on;
+* auto-shielding is always off;
+* the user specifies a policy...
+
+Auto-shielding MUST be one of:
+
+* "must opt in or out" (zcashd will do this -- i.e. refuse to start unless the option
+  is configured), or
+* always on.
+
+
+Auto-migration
+--------------
+
+
+Information leakage for transfers between pools
+-----------------------------------------------
+
+
+If no auto-migration, if you can satisfy a transfer request to Sapling from your
+Sapling funds, do so.
+
+The user's consent is needed to reveal amounts publically (as opposed
+to revealing them to the holder of a viewing key authorized to see that
+amount). Therefore, there should be per-transaction opt-in for any
+transfer that publically reveals amounts on chain.
+
+* there may be a compatibility issue for amount-revealing cross-pool txns that were
+  previously allowed without opt-in
+
+Wallets MUST NOT automatically combine funds across pools to satisfy a transfer
+(since that could reveal the total funds the user holds in some pool).
```

Comment:
It could, but as I point out earlier, a wallet that seek to maintain some proportional balance between the two pools may want to distribute ZEC from two pools just to maintain the desired balance between them, even if either pool had adequate funds for the whole spend. A UA with receivers in both shielded pools make it possible to always pay without turning any turnstile. This is very valuable for privacy.
IMO then, it's perfectly reasonable to combine pools for a single spend. If folks want to read into that to think I've drained one of those pools in my wallet, they're making stuff up.

Reply:
MUST NOT is probably too strong here; SHOULD NOT with this rationale might be better. The problem that this is trying to address is the possibility that someone sends an identifiable Sapling amount to a user; then, if that user's wallet has auto-migration turned on, that identifiable amount could be spent & revealed to the adversary, linking the targeted user to the transaction spending those funds.

Reply:
Can you elaborate on that more? You mention auto-migration as an element. So... is it that the adversary sends X ZEC to a sapling address, then waits for that X ZEC to cross the turnstile to orchard during auto-migration? If so, I don't understand how that gives the adversary information about who the targeted user is. 
I also don't see what that has to do with not spending funds from two pools at once does. 
But if I assume _no_ auto-migration, and if I imagine the victim is _paying_ the adversary in some agreed upon transaction, then yes, I suppose if a very unique amount is given from adversary to victim up front, and that same amount comes from one pool while the balance comes from another pool in the transaction, then I guess the adversary has now 'learned' that their victim likely owns the address they sent ZEC to. That seems somewhat contrived though. Possible, yes. But the adversary would have to already have a really good idea about the address correlation with one of their 'customers' that would likely send it back to him/her.